swarm/api/routes/db.py

```
@@ -242,6 +242,8 @@ async def get_db_stats():
 
         # Query counts from each table
         def safe_count(table: str) -> int:
+            if table not in {"runs", "steps", "tool_calls", "file_changes", "events", "facts"}:
+                return 0
             try:
                 result = conn.execute(f"SELECT COUNT(*) FROM {table}").fetchone()
```

Ruff (0.15.6)

[error] Possible SQL injection vector through string-based query construction

(S608)